service/routes.py

```diff
@@ -18,17 +18,142 @@
 def index():
     """ Root URL response """
     return (
-        "Reminder: return some useful information in json format about the service here",
+        "Welcome to the Shopcarts service",
         status.HTTP_200_OK,
     )
 
 
+######################################################################
+# REST API
+######################################################################
+
+# -----------------------------------------------------------
+# Create shopcarts
+# -----------------------------------------------------------
+@app.route("/shopcarts", methods=["POST"])
+def create_shopcarts():
+    """Creates a new shopcart and stores it in the database
+    Args:
+        user_id (str): the user_id of the shopcart to create
+    Returns:
+        dict: the shopcart and it's value
+    """
+    app.logger.info("Request to Create shopcart")
+    check_content_type("application/json")
+
+    # See if the shopcart already exists and send an error if it does
+    shopcart = Shopcarts()
+    shopcart.deserialize(request.get_json())     # check the fields are correct
+    user_id = request.get_json()["user_id"]
+    shopcart = Shopcarts.find_by_user_id(user_id)
+    if len(shopcart.all()) != 0:
+        abort(status.HTTP_409_CONFLICT, f"Shopcart {user_id} already exists")
+
+    # Create the new shopcart
+    shopcart = Shopcarts(user_id=user_id)
+    shopcart.create()
+
+    # Set the location header and return the new shopcart
+    location_url = url_for("read_a_shopcart", user_id=user_id, _external=True)
+    return (
+        jsonify(shopcart.serialize()),
+        status.HTTP_201_CREATED,
+        {"Location": location_url},
+    )
+
+# -----------------------------------------------------------
+# Read a shopcart
+# -----------------------------------------------------------
+@app.route("/shopcarts/<user_id>", methods=["GET"])
+def read_a_shopcart(user_id):
+    """Read a shopcart
+    Args:
+        user_id (str): the user_id of the shopcart to create
+    Returns:
+        dict: the shopcart and it's value
+    """
+    app.logger.info(f"Request to Read shopcart {user_id}...")
+
+    # See if the shopcart already exists and send an error if it does
+    shopcarts = Shopcarts.find_by_user_id(user_id).all()
+    if len(shopcarts) == 0:
+        abort(status.HTTP_404_NOT_FOUND, f"Shopcart with id '{user_id}' was not found.")
+
+    # Return the new shopcart
+    app.logger.info("Returning shopcart: %s", user_id)
+    return jsonify(shopcarts[0].serialize()), status.HTTP_200_OK
+
+######################################################################
+# LIST ALL SHOPCARTS
+######################################################################
+@app.route("/shopcarts", methods=["GET"])
+def list_shopcarts():
+    """Returns all of the Shopcarts"""
```

Comment:
Can you add args and returns like other functions?

Reply:
Got it. I'll raise a new PR for this